### DISCUSSION TITLE
When I use @babel/parser to parse other people's code, I occasionally run into parsing errors,  such as VarRedeclaration, UnexpectedToken, etc.

### DISCUSSION THREAD

**Discussion by Losstie** [babel/babel#15049] [Q&A] [answered]
When I use @babel/parser to parse other people's code, I occasionally run into parsing errors,  such as VarRedeclaration, UnexpectedToken, etc.
But as far as I know, these codes can compile and run normally in the project. So how can I parse the code without error?

For example, the code looks like this:

```
import * as _ from 'lodash';
import { observable, action, computed } from 'mobx';

// eslint-disable-next-line @typescript-eslint/naming-convention
export interface searchSubscriptionItem {
  subscriptionUnionId?: string; //
  id?: string;
  name: string; 
}

export default class SearchSubscriptionStore {
  @observable searchSubscriptionList: searchSubscriptionItem[] = [];

  @observable systemSearchSubscriptionList: searchSubscriptionItem[] = [];


  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
  @observable currentSubscription: searchSubscriptionItem = <searchSubscriptionItem>{};

  @computed
  get setAllList() {
    return [...this.systemSearchSubscriptionList, ...this.normalSearchSubscriptionList];
  }

  @action
  reInit() {
    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
    this.currentSubscription = <searchSubscriptionItem>{};
  }
}
```

and My code look like this:
```
ast = parser.parse(code, {
      sourceType: "module",
    //   errorRecovery: true,
      createParenthesizedExpressions:true,
      plugins: ["typescript",  "decorators-legacy", "jsx", "doExpressions"],
    });
```

**Comment by nicolo-ribaudo** [comment, marked as answer]
I see `<searchSubscriptionItem>{}` in that code: that syntax is only valid when the `jsx` plugin is disabled. You can test it on the TS playground too: https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wPpwGZQiOAcgBsIATNAC0IG4AoUSWOAbzggCNUBTKAN2Qdi3ADQoAxjGAQAdmPH4wAVxjcycAL448BQiE4APWnToB6U3G6piwGTAC0ZYKkHD7M7gYc2PcAAIwAJ5gVuJQwGAOVj4wpjLIILYA5vYKMnzcdtIydJ5M8LaqUNjI4txwPMhQ4pQAykpcYRFSsgCSqgQsdHAVDahNkdkAqjLZrWQA-ABcFTDhMkk0cObdcMCTM6hzyfQ98SDcm9sLS3QaJnnQ8GTcJUrE8OLEaOi13FU19Y3hg7K1MNByl0en5ODx+K5ypVqnU+gMWjIADLOGCbd4wr79H4I9rcEAAbQAunAALxwIn0Vagri8ARCKGBLZ4t4fWHfZrZZFbNGszHwsYdImk8mEymrcyWayFRzOSHuTzeWzlALBULY+zRQqmNKoFGZBxBEL2F68BGoKlg2mQuDiJRQKD6vnY7I8jFw51tDrCgA80M+7o5nrxAD4WBpKSCFOAVGpVkluPAeDAAILEYhcmAACgAlKxVj0HTA7TJyQA6cswSjOUuoRkdFlu9m-JEosTl0uV6syaAgZDEBv+psIjOi1bnKmlBGrB2tUZZ3PAnrLCyauwylz0+Veew+ZWGtXNDVSuza2S6plr-fG1Dgs35uCd1Cl232x0B5s+v1srGBmS4kChuGY5nHQQA

**Reply by Losstie** [reply]
thanks, I got it.  It seems that the above files and jsx files cannot be parsed with one set of code.
